sdv/sampling/hierarchical_sampler.py

```diff
@@ -241,7 +241,13 @@ def _finalize(self, sampled_data):
             synthesizer = self._table_synthesizers.get(table_name)
             dtypes = synthesizer._data_processor._dtypes
             for name, dtype in dtypes.items():
-                table_rows[name] = table_rows[name].dropna().astype(dtype)
+                try:
+                    table_rows[name] = table_rows[name].dropna().astype(dtype)
+                except Exception:
+                    LOGGER.info(
+                        "Could not cast back to column's original dtype, keeping original typing."
+                    )
+                    table_rows[name] = table_rows[name].dropna()
```

Comment:
let's add unit tests for this as well

Reply:
Sounds good, added both unit test for this try and catch and an integration test.